src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'green' | 'red' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
 
-export class Circle {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || a + b <= c
+      || a + c <= b
+      || b + c <= a
+    ) {
+      throw new Error(
+        'Check the size of sides, they must be positive'
+        + 'or sides don\'t satisfy the triagle condition',
+      );
+    }
+  }
 
+  getArea(): number {
+    const s:number = (this.a + this.b + this.c) / 2;
+    const square: number
+      = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
```

Comment:
Типізація тут зайва, якщо вираз завжди повертає number, ця змінна автоматично отримає тип number

```suggestion
    const s = (this.a + this.b + this.c) / 2;
    const square = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
```